crypto/evp/evp_rand.c

```diff
@@ -0,0 +1,558 @@
+    }
+    if (parent != NULL) {
+        if (!EVP_RAND_CTX_enable_locking(parent)) {
+            EVPerr(0, EVP_R_UNABLE_TO_ENABLE_PARENT_LOCKING);
+            OPENSSL_free(ctx);
+            return NULL;
+        }
+        parent_ctx = parent->data;
+        parent_dispatch = parent->meth->dispatch;
+    }
+    if ((ctx->data = rand->newctx(ossl_provider_ctx(rand->prov), parent_ctx,
+                                  parent_dispatch)) == NULL
+            || !EVP_RAND_up_ref(rand)) {
+        EVPerr(0, ERR_R_MALLOC_FAILURE);
+        rand->freectx(ctx->data);
+        OPENSSL_free(ctx);
+        return NULL;
+    }
+    ctx->meth = rand;
+    return ctx;
+}
+
+void EVP_RAND_CTX_free(EVP_RAND_CTX *ctx)
+{
+    if (ctx != NULL) {
+        ctx->meth->freectx(ctx->data);
+        ctx->data = NULL;
+        EVP_RAND_free(ctx->meth);
+        OPENSSL_free(ctx);
+    }
+}
+
+EVP_RAND *EVP_RAND_CTX_rand(EVP_RAND_CTX *ctx)
+{
+    return ctx->meth;
+}
+
+int EVP_RAND_CTX_get_params(EVP_RAND_CTX *ctx, OSSL_PARAM params[])
+{
+    int res = 1;
+
+    if (ctx->meth->get_ctx_params != NULL) {
+        if (!evp_rand_lock(ctx))
+            return 0;
+        res = ctx->meth->get_ctx_params(ctx->data, params);
+        evp_rand_unlock(ctx);
+    }
+    return res;
+}
+
+int EVP_RAND_CTX_set_params(EVP_RAND_CTX *ctx, const OSSL_PARAM params[])
+{
+    int res = 1;
+
+    if (ctx->meth->set_ctx_params != NULL) {
+        if (!evp_rand_lock(ctx))
+            return 0;
+        res = ctx->meth->set_ctx_params(ctx->data, params);
+        evp_rand_unlock(ctx);
+        /* Clear out the cache state because the values can change on a set */
+        ctx->strength = 0;
+        ctx->max_request = 0;
+    }
+    return res;
+}
+
+const OSSL_PARAM *EVP_RAND_gettable_params(const EVP_RAND *rand)
+{
+    return rand->gettable_params == NULL ? NULL : rand->gettable_params();
+}
+
+const OSSL_PARAM *EVP_RAND_gettable_ctx_params(const EVP_RAND *rand)
+{
+    return rand->gettable_ctx_params == NULL ? NULL
+                                             : rand->gettable_ctx_params();
+}
+
+const OSSL_PARAM *EVP_RAND_settable_ctx_params(const EVP_RAND *rand)
+{
+    return rand->settable_ctx_params == NULL ? NULL
+                                             :rand->settable_ctx_params();
+}
+
+void EVP_RAND_do_all_provided(OPENSSL_CTX *libctx,
+                              void (*fn)(EVP_RAND *rand, void *arg),
+                              void *arg)
+{
+    evp_generic_do_all(libctx, OSSL_OP_RAND,
+                       (void (*)(void *, void *))fn, arg,
+                       evp_rand_from_dispatch, evp_rand_free);
+}
+
+void EVP_RAND_names_do_all(const EVP_RAND *rand,
+                           void (*fn)(const char *name, void *data),
+                           void *data)
+{
+    if (rand->prov != NULL)
+        evp_names_do_all(rand->prov, rand->name_id, fn, data);
+}
+
+int EVP_RAND_CTX_instantiate(EVP_RAND_CTX *ctx, unsigned int strength,
```

Comment:
This isn't a 'EVP_RAND_CTX` constructor/destructor or direct getter/setter, so should be named `EVP_RAND_instantiate`.

I see that you've made the "just use the type of the first argument" exercise with a number of other similar functions, something I've strongly opposed to.  In essence, that means that no matter what, there's a function rename fest waiting to happen, might as well bite that bullet now.

Reply:
Yeah, the sed magic fixes this :)
There are a couple of calls I'm unsure about.

Reply:
> There are a couple of calls I'm unsure about.

I look forward to discussing it in the follow up PR :wink: